src/util/evented.js

```diff
@@ -6,6 +6,7 @@ type Listener = (Object) => any;
 type Listeners = { [string]: Array<Listener> };
 
 function _addEventListener(type: string, listener: Listener, listenerList: Listeners) {
+    _removeEventListener(type, listener, listenerList);
```

Comment:
Rather than remove and re-add, make this an affirmative check for the current presence of listener, so that a redundant `on` doesn't change the relative order of listeners. E.g. given `map.on('click', listenerA); map.on('click', listenerB); map.on('click', listenerA);`, `listenerA` should be the first one called.